### ISSUE TITLE
Secret Chat Id 

### ISSUE THREAD

**Issue by yepotor374** [tdlib/td#1095]
So after creating a secret chat, in getChats I loop through the Id's returned by getChats and filter the secretChat using ChatTypeSecret.

Thing is the returned chat is just a normal chat with a normal ID.
The Id is good enough to send messages to the secret chat but is still not the true secret_chat_id.

How do you obtain secret_chat_id from the chat object returned from getChats.

Thanks!

**Comment by levlam** [comment]
`ChatTypeSecret` contains `secret_chat_id` field.

**Comment by yepotor374** [comment]
Ok now i feel very stupid.

Thanks Levlam

**Comment by yepotor374** [comment]
Also i see all objects have a "Extra" string variable, can I use that to carry secret_chat_id in the chat object?

**Comment by levlam** [comment]
No. See https://core.telegram.org/tdlib/getting-started#tdlib-interface for description of the @extra field.

**Comment by yepotor374** [comment]
So on update of Update.UpdateSecretChat it passes the secretchat, how would i find the corresponding tdapi.chat Id to clear it in DeleteChatHistory since it will not except secretchat id

**Comment by levlam** [comment]
You can save correspondence chat.id <-> secret_chat_id.

**Comment by yepotor374** [comment]
ok so save it locally? otherwise there it no way of correlating the two.

**Comment by levlam** [comment]
Save in memory.

**Comment by levlam** [comment]
Or better use createSecretChat to get the chat by secret_chat_id.

**Comment by yepotor374** [comment]
thanks
